crates/sui-indexer-alt-framework/src/ingestion/client.rs

```diff
@@ -148,8 +149,7 @@ impl IngestionClient {
                     FetchData::Raw(bytes) => {
                         self.metrics.total_ingested_bytes.inc_by(bytes.len() as u64);
                         Blob::from_bytes(&bytes).map_err(|e| {
-                            self.metrics.inc_retry(
-                                checkpoint,
+                            self.metrics.inc_permanent(
```

Comment:
I think this should still be a retryable error -- you're right that attempting to deserialize the same bytes over and over again will just keep failing, but when we retry we also redo the fetch, so in case deserialization failed because the bytes were corrupted during transport, we will recover.